edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookFrameworkServiceImpl.java

```diff
@@ -188,6 +188,60 @@ public Object doInHibernate(Session session) throws HibernateException {
 		});
 	}
 
+	public void updateGradeMapping(final Object gradeMapping) {
+		getHibernateTemplate().execute(new HibernateCallback() {
+			public Object doInHibernate(Session session) throws HibernateException {
+				session.update((GradeMapping)gradeMapping);
+				session.flush();
+				return null;
+			}
+		});
+	}
+
+	public void saveGradeMappingToGradebook(String scaleUuid, Object gradebook) {
+		getHibernateTemplate().execute(new HibernateCallback() {
+			public Object doInHibernate(Session session) throws HibernateException {
+
+				List gradingScales = session.createQuery("from GradingScale as gradingScale where gradingScale.unavailable=false").list();
+
+
+				for (Iterator iter = gradingScales.iterator(); iter.hasNext(); ) {
+					GradingScale gradingScale = (GradingScale) iter.next();
+					if (gradingScale.getUid().equals(scaleUuid)){
+						GradeMapping gradeMapping = new GradeMapping(gradingScale);
+						gradeMapping.setGradebook((Gradebook)gradebook);
+						session.save(gradeMapping);
+					}
+				}
+				session.flush();
+				return null;
+			}
+		});
+	}
+
+	public List getAvailableGradingScales() {
+
+		List gradingScales2 = (List)getHibernateTemplate().execute(new HibernateCallback() {
```

Comment:
Could you call this 'rval' or something? It's not immediately obvious why its '2'.